kernel/src/actions/mod.rs

```diff
@@ -120,27 +127,67 @@ impl Metadata {
 pub struct Protocol {
     /// The minimum version of the Delta read protocol that a client must implement
     /// in order to correctly read this table
-    pub min_reader_version: i32,
+    min_reader_version: i32,
```

Comment:
did we decide that making these not pub was okay

Reply:
well we have `pub` getters so I think semantically it's the same, just might have to tell someone downstream to add some parenthesis